src/api-binder.ts

```diff
@@ -580,9 +587,25 @@ export class APIBinder {
 				this.deviceState.triggerApplyTarget({ force, isFromApi });
 			}
 		})
-			.tapCatch(err => {
-				log.error(`Failed to get target state for device: ${err}`);
+			.tapCatch(ContractValidationError, ContractViolationError, e => {
+				log.error(`Could not store target state for device: ${e}`);
+				this.logger.logSystemMessage(
+					`Could not move to new release: ${e.message}`,
+					{},
+					'targetStateRejection',
+					false,
+				);
 			})
+			.tapCatch(
+				(e: unknown) =>
+					!(
+						e instanceof ContractValidationError ||
+						e instanceof ContractViolationError
```

Comment:
If the above `.tapCatch` was a `.catch`, there would be no need for filtering on `e`'s class here.

Reply:
The thing is we still want these values to be raised up to the caller (device state) which will cause things to back off. I couldn't think of a nice way to do it really.